[Response Ops][Rule Form V2] Rule form v2: Actions Modal and Actions Form

JiaweiWu opened the pull request:
## Summary
Issue: https://github.com/elastic/kibana/issues/179105
Related PR: https://github.com/elastic/kibana/pull/180539

Final PR of the rule actions V2 PR (2/2 of the actions PRs). This PR contains the actions modal and actions form. This PR depends on https://github.com/elastic/kibana/pull/186490. 

I have also created a example plugin to demonstrate this PR. To access:

1. Run the branch with yarn start --run-examples
2. Navigate to http://localhost:5601/app/triggersActionsUiExample/rule/create/<ruleTypeId> (I use .es-query)
3. Create a rule
4. Navigate to http://localhost:5601/app/triggersActionsUiExample/rule/edit/<ruleId> with the rule you just created to edit the rule

<img width="1236" alt="Screenshot 2024-07-02 at 5 15 51 PM" src="https://github.com/elastic/kibana/assets/74562234/1dc5d2a9-804a-4861-94ba-814de73dc3ab">

![Screenshot 2024-07-08 at 10 53 44 PM](https://github.com/elastic/kibana/assets/74562234/07efade1-4b9c-485f-9833-84698dc29219)

<img width="1087" alt="Screenshot 2024-07-02 at 5 15 58 PM" src="https://github.com/elastic/kibana/assets/74562234/903e66b5-f9a1-4d09-b121-b1dcecdff72c">


### Checklist
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios


<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

elasticmachine commented:
Pinging @elastic/response-ops (Team:ResponseOps)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/219261)
* Commit: 6c99a7ad556ddad83a977328462ee502009c787d

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/219261#019075f5-d794-455a-824d-507cdb578e5e)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/219261","number":219261,"commit":"6c99a7ad556ddad83a977328462ee502009c787d"}],"number":219261}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/220257)
* Commit: a40074f63d1ba2ab4d34b0d771d008cccca2f716

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/220257#019095e4-0f92-471b-8f00-4a868ec74f40)

### History
* :broken_heart: [Build #219261](https://buildkite.com/elastic/kibana-pull-request/builds/219261) failed 6c99a7ad556ddad83a977328462ee502009c787d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220257","number":220257,"commit":"a40074f63d1ba2ab4d34b0d771d008cccca2f716"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/219261","number":219261,"commit":"6c99a7ad556ddad83a977328462ee502009c787d"}],"number":220257}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/220499)
* Commit: 06b35a5ddb0976caa2904f31cf388480e5778a89

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/220499#01909a3c-0624-4f78-b46f-564a9adac572)

### History
* :broken_heart: [Build #220257](https://buildkite.com/elastic/kibana-pull-request/builds/220257) failed a40074f63d1ba2ab4d34b0d771d008cccca2f716
* :broken_heart: [Build #219261](https://buildkite.com/elastic/kibana-pull-request/builds/219261) failed 6c99a7ad556ddad83a977328462ee502009c787d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220499","number":220499,"commit":"06b35a5ddb0976caa2904f31cf388480e5778a89"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220257","number":220257,"commit":"a40074f63d1ba2ab4d34b0d771d008cccca2f716"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/219261","number":219261,"commit":"6c99a7ad556ddad83a977328462ee502009c787d"}],"number":220499}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/220502)
* Commit: b09954f5ab33778398bb13a5882df158d0720075

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/220502#01909a8b-9318-4c7f-9a94-09ce86f6c3a0)

### History
* :broken_heart: [Build #220499](https://buildkite.com/elastic/kibana-pull-request/builds/220499) failed 06b35a5ddb0976caa2904f31cf388480e5778a89
* :broken_heart: [Build #220257](https://buildkite.com/elastic/kibana-pull-request/builds/220257) failed a40074f63d1ba2ab4d34b0d771d008cccca2f716
* :broken_heart: [Build #219261](https://buildkite.com/elastic/kibana-pull-request/builds/219261) failed 6c99a7ad556ddad83a977328462ee502009c787d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220502","number":220502,"commit":"b09954f5ab33778398bb13a5882df158d0720075"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220499","number":220499,"commit":"06b35a5ddb0976caa2904f31cf388480e5778a89"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220257","number":220257,"commit":"a40074f63d1ba2ab4d34b0d771d008cccca2f716"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/219261","number":219261,"commit":"6c99a7ad556ddad83a977328462ee502009c787d"}],"number":220502}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/222510)
* Commit: 919c6d2e379d2a63f2cdbd8969ff64db51eb3552

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/222510#0190c84e-1668-48d3-9507-4762c58c2035)

### History
* :broken_heart: [Build #220502](https://buildkite.com/elastic/kibana-pull-request/builds/220502) failed b09954f5ab33778398bb13a5882df158d0720075
* :broken_heart: [Build #220499](https://buildkite.com/elastic/kibana-pull-request/builds/220499) failed 06b35a5ddb0976caa2904f31cf388480e5778a89
* :broken_heart: [Build #220257](https://buildkite.com/elastic/kibana-pull-request/builds/220257) failed a40074f63d1ba2ab4d34b0d771d008cccca2f716
* :broken_heart: [Build #219261](https://buildkite.com/elastic/kibana-pull-request/builds/219261) failed 6c99a7ad556ddad83a977328462ee502009c787d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222510","number":222510,"commit":"919c6d2e379d2a63f2cdbd8969ff64db51eb3552"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220502","number":220502,"commit":"b09954f5ab33778398bb13a5882df158d0720075"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220499","number":220499,"commit":"06b35a5ddb0976caa2904f31cf388480e5778a89"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220257","number":220257,"commit":"a40074f63d1ba2ab4d34b0d771d008cccca2f716"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/219261","number":219261,"commit":"6c99a7ad556ddad83a977328462ee502009c787d"}],"number":222510}
buildkite-pr-comment-->

cnasikas reviewed (commented):
Great work! I tested the PR and I found the following bugs. I will do the code review on a second pass.

- Permissions: If I view the form without permissions to connectors react query refetches the queries 3 times, the loading spinner is shown, and then it shows no connectors. How do you think we should handle 403 errors? In cases we show a message to the user that she does no have permissions to use connectors.

https://github.com/user-attachments/assets/0514869a-7d5b-4423-8192-1e53d18c612c

<img width="1082" alt="Screenshot 2024-07-24 at 9 28 52 AM" src="https://github.com/user-attachments/assets/09fd2cb7-95b6-472e-8d89-cb7a20772bb0">

<img width="1216" alt="Screenshot 2024-07-24 at 9 37 05 AM" src="https://github.com/user-attachments/assets/e9407709-e3a8-4cab-ba59-62bb0f5c090e">

- If I go to the flyout the padding around the connector's car is removed

<img width="607" alt="Screenshot 2024-07-24 at 9 39 34 AM" src="https://github.com/user-attachments/assets/cf8139fa-e099-458a-bde7-3c1a88b1bdc1">

- When I load a rule with the wrong ID I get the following error. Should we change the text from `rule type` -> `rule`?

<img width="549" alt="Screenshot 2024-07-24 at 9 43 19 AM" src="https://github.com/user-attachments/assets/84976e23-38bc-40ff-8e94-b0faaa6f541b">

- If I press the "Create rule" button the connector's modal appears to confirm my choice (no connectors in the rule). The rule is created behind the scenes without waiting for me to press "Save rule" in the modal.

https://github.com/user-attachments/assets/a833a0c7-92f5-4cc1-a9ce-f2cf3a344efa

- Some actions have required fields. It seems that I can save the rule without having to fill the required fields. The request will contain the action but the `params` would be an empty object (btw should not the API throw an error?).

https://github.com/user-attachments/assets/a61a5960-5ce2-4897-b70d-a7534b6117d2

<img width="731" alt="Screenshot 2024-07-24 at 9 50 05 AM" src="https://github.com/user-attachments/assets/57b9da25-113d-4db2-afd7-21e8cbc5b842">

- If I close the connector's accordition I get a validation error even though I did not interacted with the connector. I checked and this is the current behavior on the flyout. Should we change the user's experience and show the error only a) if the user interacts with the form or b) the user pressed the "Save rule" button. Not for this PR ofc :). Wdyt?

cnasikas commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_item.test.tsx:
Should we add tests that test all the logic with the default/aad params? Tbh with you I do not understand the logic with the params that much 🙂.

JiaweiWu replied:
i dont either TBH i'll add it later when we start testing it

cnasikas commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_message.tsx:
```diff
@@ -0,0 +1,118 @@
+      <EuiFlexGroup direction="column" data-test-subj="ruleActionsMessage">
+        <EuiFlexItem>
+          <Suspense fallback={null}>
+            <ParamsFieldsComponent
```
This PR https://github.com/elastic/kibana/pull/188724 fixed a bug we had (missing of legacy EUI provider) that made some rule types crash. Do you mind if you apply the same fix here until o11y migrates its rule types from styled components to emotion?

cnasikas commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_settings.tsx:
```diff
@@ -0,0 +1,263 @@
+    intervalUnit,
+  ] ?? [-1, 's'];
+
+  const [showMinimumThrottleWarning, showMinimumThrottleUnitWarning] = useMemo(() => {
```
What do you think of moving this logic to a function (inside the same file)?

cnasikas commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_settings.tsx:
```diff
@@ -0,0 +1,263 @@
+                  featureIds={[
+                    producerId === 'alerts' ? 'stackAlerts' : (producerId as ValidFeatureId),
+                  ]}
+                  appName="stackAlerts"
```
Is it possible for this component to be used outside the stack page?

JiaweiWu replied:
Yes, I (or someone) will make these dynamic when we do so

cnasikas commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_system_actions_item.tsx:
```diff
@@ -0,0 +1,274 @@
+                </EuiBadge>
+              </EuiFlexItem>
+            )}
+            {actionTypeModel.isExperimental && (
```
I noticed that we did not provide the same functionality for normal actions. Should we add it?

cnasikas reviewed (commented):
This is a great PR. I can not believe how much clean at least the rule form became. I left some questions and I wanted to discuss the following:

- I see that the `index` is passed in some components but given now that we switched to `action.uuid` I do not think is needed. What do you think of removing it? I know that the connectors' params still using it. To avoid changing all connectors maybe we can fake it to `0` and ignore it in the framework level.
- I am a bit confused with the purpose of `onUseDefaultMessage`. What is the purpose of it? Could we remove it?
- The normal actions and the system actions have, correctly, different components. They share common UI but a good amount of different logic. If in the future we would need to apply a UI fix or enhance it we would have to remember to do it in both places. Is it feasible to do something like `<CommonRuleActionItem><RuleActionItem /></CommonRuleActionItem>` where the `CommonRuleActionItem` has the common UI and then each `*Item` contains its own logic?
- I noticed that the following files miss tests:
    - packages/kbn-alerts-ui-shared/src/rule_form/utils/has_fields_for_aad.ts
    - packages/kbn-alerts-ui-shared/src/rule_form/utils/get_selected_action_group.ts
    - packages/kbn-alerts-ui- 
    - shared/src/rule_form/validation/validate_params_for_warnings.ts

JiaweiWu commented:
@cnasikas For the case where the user doesn't have privilege to actions and connectors, i made the following changes, it should match with existing behaviour, if the user doesn't have access to connectors they can still create a rule but they cant add actions. If they're trying to edit a rule with connectors then they simply cannot

## Create rule
<img width="1238" alt="Screenshot 2024-09-16 at 11 10 30 PM" src="https://github.com/user-attachments/assets/406e0cd6-d60a-4a88-9a04-c34e35fc2d48">

## Edit rule
<img width="1151" alt="Screenshot 2024-09-16 at 11 09 55 PM" src="https://github.com/user-attachments/assets/a6ac6aea-f5d4-4f5c-ac25-e8724d16a1cf">

JiaweiWu commented:
https://github.com/elastic/kibana/pull/187434/commits/e0fab84b03c12318746dc8f1a2b70974b3ac9a5f

Fixes/changes:

- Fixes styling issues in the existing flyout
- Fixes double saving when creating a rule
- Changed insufficient action permissions behaviour to mimic existing flyout
- Swapped "Messages" and "Settings" tab so the action params get validated automatically

JiaweiWu commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_alerts_filter.tsx:
```diff
@@ -1,55 +1,69 @@
+    notifications: { toasts },
+    unifiedSearch,
+    dataViews,
+  } = plugins || propsPlugins || {};
```
They are required, not sure why i have the `|| {}`, going to remove it

JiaweiWu commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_alerts_filter.tsx:
```diff
@@ -1,55 +1,69 @@
-    if (!deepEqual(state, nextState)) onChange(nextState);
-  }, [queryEnabled, query, state, onChange]);
+  const toggleQuery = useCallback(() => {
+    onChange(query ? undefined : DEFAULT_QUERY);
```
Well we use the `query` prop to determine if the toggle is on or off, so essentially it's doing what you're describing

JiaweiWu commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_connectors_modal.tsx:
```diff
@@ -0,0 +1,348 @@
+        preconfiguredConnectors
+      );
+
+      if (!actionType.enabledInConfig && !checkEnabledResult.isEnabled) {
```
Probably, but I'm just copying the old logic: `x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_form.tsx` line 313

JiaweiWu commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_connectors_modal.tsx:
```diff
@@ -0,0 +1,348 @@
+}
+
+export const RuleActionsConnectorsModal = (props: RuleActionsConnectorsModalProps) => {
+  const { connectors, actionTypes, onClose, onSelectConnector } = props;
```
The connector map isnt actually a map of the connector types but it's a counter of how many of the same connectors there are for the facet buttons. 

<img width="393" alt="image" src="https://github.com/user-attachments/assets/77aecc43-44bf-49fd-9ed3-70c8ffaad17f">

yea seems like a super small optimization im not sure if it's worth the extra, maybe we can come back to it if it's a problem

JiaweiWu commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_connectors_modal.test.tsx:
```diff
@@ -0,0 +1,316 @@
+
+    userEvent.clear(screen.getByTestId('ruleActionsConnectorsModalSearch'));
+
+    userEvent.type(screen.getByTestId('ruleActionsConnectorsModalSearch'), 'doesntexist');
```
isnt this the 1?

JiaweiWu commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_item.tsx:
```diff
@@ -0,0 +1,510 @@
+      aadParams: Record<string, SavedObjectAttribute>,
+      params: Record<string, SavedObjectAttribute>
+    ) => {
+      if (isEmpty(aadParams) && action.params.subAction) {
```
Not too sure, i'm guessing we want to use subAction if it exists, if not default to AAD params

JiaweiWu commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_item.tsx:
```diff
@@ -0,0 +1,510 @@
+  const dispatch = useRuleFormDispatch();
+  const actionTypeModel = actionTypeRegistry.get(action.actionTypeId);
+  const actionType = actionTypes.find(({ id }) => id === action.actionTypeId)!;
+  const connector = connectors.find(({ id }) => id === action.id)!;
```
I do a check for it at the top level where the API calls are made

JiaweiWu commented on packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_item.tsx:
```diff
@@ -0,0 +1,510 @@
+  const [useDefaultMessage, setUseDefaultMessage] = useState(false);
+
+  const [storedActionParamsForAadToggle, setStoredActionParamsForAadToggle] = useState<
+    Record<string, SavedObjectAttribute>
```
Can't do this sadly because i'll have to change the `action` and `connectorType` types since this saves them into this state variable

JiaweiWu commented:
@cnasikas I believe this PR is ready for review now as I've addressed all of your comments

https://github.com/elastic/kibana/pull/187434/commits/0fd82653bebd428268668bc45ad900ea310abe89 changes:

- Address remaining comments
- Added tests for:
  - packages/kbn-alerts-ui-shared/src/rule_form/utils/has_fields_for_aad.ts
  - packages/kbn-alerts-ui-shared/src/rule_form/utils/get_selected_action_group.ts
  - packages/kbn-alerts-ui-shared/src/rule_form/validation/validate_params_for_warnings.ts


> I see that the index is passed in some components but given now that we switched to action.uuid I do not think is needed. What do you think of removing it? I know that the connectors' params still using it. To avoid changing all connectors maybe we can fake it to 0 and ignore it at the framework level.

Could you clarify what you mean by index? 

> I am a bit confused with the purpose of onUseDefaultMessage. What is the purpose of it? Could we remove it?

I'll be honest I'm not sure either, I will do more research on it and update the test document with my findings.

> The normal actions and the system actions have, correctly, different components. They share a common UI but a good amount of different logic. If in the future we would need to apply a UI fix or enhance it we would have to remember to do it in both places. Is it feasible to do something like <CommonRuleActionItem><RuleActionItem /></CommonRuleActionItem> where the CommonRuleActionItem has the common UI and then each *Item contains its logic?

I would agree but again, I think it's easier to have 2 for now since I'm trying to match feature parity with the existing components. I think we went cross this road we can refactor it.

JiaweiWu commented:
@elasticmachine merge upstream

cnasikas commented:
Hey @JiaweiWu! I tested and I can confirm that the bugs are fixed except for this one:

> - When I load a rule with the wrong ID I get the following error. Should we change the text from `rule type` -> `rule`?
<img width="549" alt="Screenshot 2024-07-24 at 9 43 19 AM" src="https://github.com/user-attachments/assets/84976e23-38bc-40ff-8e94-b0faaa6f541b">

Also, I noticed if we get a 404 error from the `_resolve` API, react query retries 3 times before showing the above error (screenshot). I think, if possible, it is better to not retry on 404 as nothing will change after 3 retries. We can leave this for another PR.

>Could you clarify what you mean by index?

I was talking about the `index` prop passed to the `RuleActionsItem`. Nevermind, we can clean it up on another PR. 

>Looks like it's being used by the connector params components, check here for an example:
Ok!

>I would agree but again, I think it's easier to have 2 for now since I'm trying to match feature parity with the existing components. I think we went cross this road we can refactor it.
Ok!

JiaweiWu commented:
In the commit https://github.com/elastic/kibana/pull/187434/commits/2cd1eed3115bcf7ee33371aadd78e6610f95c333:

- Fixed messaging for when the user tries to edit a rule that doesn't exist, also no longer calls `_resolve` 3 times
- Changed rule form to use `alerts` instead of `stackAlerts` for default consumer
- Added the use AAD template for mustache switch

cnasikas reviewed (approved):
I tested and all bugs are fixed! LGTM! I left some nit comments but we can address them on a different PR.

joana-cps commented:
> Thanks, @JiaweiWu! I think the pattern in Kibana is to hide the section entirely if the user does not have access to a feature (though in cases we show a similar same message). I tested on the current flyout form and it is hidden. Should we follow the same pattern? @joana-cps Wdyt? About the editing, it seems that we do not allow it if you do not have access to actions. Just curiosity (not to do anything), what are the technical constraints with allowing the users to edit the rule but not the actions?

Although I don't like to have a step in the form that is disabled, I think it is a good idea to keep the section as @JiaweiWu did and inform that the user doesn't have privileges (at least for now).
And, I agree that we should explore the use cases for the Edit because it is weird to block the full edit capability when only a part of the rule requires a different privilege.

JiaweiWu commented:
@elasticmachine merge upstream

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/11065745004

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/194254">](https://github.com/elastic/kibana/pull/194254)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
